jspsych-uil-session.js

```diff
@@ -0,0 +1,76 @@
+'use strict'
+
+if (typeof uil === 'undefined')
+    console.error("UiL Main library not loaded! " +
+                  "Refusing to load browser info functions");
+else
+{
+    if ('session' in uil) {
+        console.warn(
+            "uil.browser already exists, this is unexpected, " +
+                "we'll overwrite it."
+        );
+    }
+
+    uil.session = {};
+
+    (function (context) {
+        class API {
+            constructor(host, access_key) {
+                this.host = host;
+                this.headers = {};
+            }
+
+            _request(url, method, data) {
+                let params = {
+                    method: method,
+                };
+                if (typeof(data) !== 'undefined') {
+                    params.body = new Blob([data], {type: 'text/plain'});
+                }
+                return new Promise((resolve, reject) => {
+                    fetch(this.host + url, params).then((response) => {
+                        response.json().then(resolve).catch(reject);
+                    }).catch(reject);
+                });
+            }
+
+            get(url) {
+                return this._request(url, 'GET');
+            }
+            post(url, data) {
+                return this._request(url, 'POST', data);
+            }
+
+            sessionStart(access_key) {
+                return this.post(`${access_key}/participant/`);
+            }
+
+            sessionUpload(access_key, session_id, data) {
+                return this.post(`${access_key}/upload/${session_id}/`, data);
+            }
+        }
+
+        var session_id = null;
+        var api = new API(uil.resolveServer());
+
+        context.isActive = function() {
+            return session_id !== null;
+        }
+
+        context.start = function(access_key, callback) {
```

Comment:
The `access_key` parameter is not used inside the constructor. It would be nice to either cache the key inside an API instance, and then drop the parameter from the functions below. Or to drop the parameter from the constructor and keep it as separate parameter to the API method below.
In other parts of the jspsych we have went through some effort to store the access key as an cached variable, so perhaps it nice to do so here.